slick/slick.js

```diff
@@ -2933,6 +2942,41 @@ Issues: http://github.com/kenwheeler/slick/issues
 		return out;
 	};
 
+	// @param  {Array}|{NodeList} `collectionOfNode` A collection of node to filter
+	// @param  {function}|{cssSelector}|{Node} `filter` The filter param
+	// @return {Array} of HTMLElement filtred by `filter`
+	// @usage Slick.filterNodeUtil(collectionOfNode, (funtion(elem,index,array){...} | '.my-selector' | node) );
+	// Equivalent to jQuery.filter() method, BUT do not support extended jQuery selector https://api.jquery.com/category/selectors/jquery-selector-extensions/
+	Slick.prototype.filterNodeUtil = function(collectionOfNode, filter) {
+		var _ = this,
+			domParent = document.createElement("div"),
+			selectedNodes,
+			filterFunction;
+
+		if (typeof filter === "string" && filter) {
+			Array.prototype.forEach.call(collectionOfNode, function(elem) {
+				domParent.appendChild(elem);
+			});
+			selectedNodes = _.queryAll(filter, domParent);
+			filterFunction = function(elem) {
+				return selectedNodes.lastIndexOf(elem) >= 0;
+			};
+		} else if (filter instanceof HTMLElement) {
+			filterFunction = function(elem) {
+				return filter === elem;
+			};
+		} else if (typeof filter === "function") {
+			filterFunction = filter;
+		} else {
+			filterFunction = function() {
+				return true;
+			};
+		}
+
+		return Array.prototype.slice.call(collectionOfNode).filter(filterFunction);
```

Comment:
What about using `return Array.prototype.filter.call(collectionOfNode, filterFunction);`? It will be more concise.